### PR TITLE
add setOwner support

### PR DESCRIPTION
**Pull request by jamesarosen**
See https://github.com/rwjblue/ember-getowner-polyfill/pull/6#pullrequestreview-11636408

### CODE REVIEW

File: vendor/install-getowner-polyfill.js [hunk 1]
```diff
@@ -135,5 +135,17 @@
         };
       }
     });
+
+    Object.defineProperty(_Ember, 'setOwner', {
```
Comment (rwjblue) [inline vendor/install-getowner-polyfill.js]: Can you move this and guard this in `if (!_Ember.setOwner) {` (it is currently inside of the `if (!_Ember.getOwner) {` guard).

Comment (jamesarosen) [inline vendor/install-getowner-polyfill.js]: Done

**Comment by jamesarosen** [comment]
The failing test is because [ember-test-helpers calls `setOwner` with `container.owner`](https://github.com/emberjs/ember-test-helpers/blob/master/lib/ember-test-helpers/test-module.js#L121-L123), but this polyfill doesn't tell the `container` that its `owner` is the `FakeOwner`.

Some ideas:

 * change the `getOwner` and `setOwner` polyfills to set `object.container.owner`. It might be good to also change `getOwner` to check for `object.container.owner` before creating a new `FakeOwner`
 * change the `setOwner` to no-op when `owner` is `null`, though this is different from the real implementation, which allows setting to `null`

**Comment by rwjblue** [comment]
@mmun - OK, CI is happy, the tests look good, any objections to landing?

File: vendor/install-getowner-polyfill.js [hunk 2]
```diff
@@ -153,19 +158,38 @@
...
+        var owner = new FakeOwner(object);
+        container[OWNER] = owner;
+      }
+
+      return container[OWNER];
+    }
+
+    function setOwner(object, owner) {
+      object[OWNER] = owner;
+    }
```
Comment (mmun) [inline vendor/install-getowner-polyfill.js:180]: This probably belongs in the other branch, although it may not matter with hoisting rules...

**Pull request merged**